python/paddle/distributed/auto_parallel/api.py

```diff
@@ -90,9 +91,263 @@ def sharding_specs(self):
 
 
+class DistModel:
+    """
+    DistModel is a wrapper of the network model for the use of static mode
+    auto parallel. DistModel contains the distributed Graph of the model and
+    offers the APIs for training, evaluation and prediction.
+
+    Please first set the DistModel to "train", "eval" or "predict" mode and
+    then use the __call__ method for training, evaluation and prediction
+    respectively.
+
+    In "train" mode, executing ``__call__`` will update the parameters
+    of the model and return the loss. In "eval" mode, executing ``__call__``
+    will return the loss. In "predict" mode, executing ``__call__`` returns a
+    dict that contains the outputs of the model, where the value of "out0" is
+    the first output.
+
+    DistModel is generated by ``static_decorate``, for more details of the
+    usage, please refer to the sample code in ``static_decorate``.
+    """
+
+    def __init__(
+        self, layer, loss=None, optimizer=None, strategy=None, metrics=None
+    ):
+        self._feed_name_list = []
+        self._engine = Engine(
+            layer, loss, optimizer, metrics, strategy=strategy
+        )
+        self._mode = None
+        self._feed_name_list = {}
+
+    def train(self):
+        """
+        Set the mode of DistModel to "train".
+        """
+        if not self._engine._has_prepared["train"]:
+            raise RuntimeError("The model for training has not been prepared.")
+        self._mode = "train"
+        self._engine.to_mode("train")
+
+    def eval(self):
+        """
+        Set the mode of DistModel to "eval".
+        """
+        if not self._engine._has_prepared["eval"]:
+            raise RuntimeError(
+                "The model for evaluation has not been prepared."
+            )
+        self._mode = "eval"
+        self._engine.to_mode("eval")
+
+    def predict(self):
+        """
+        Set the mode of DistModel to "predict".
+        """
+        if not self._engine._has_prepared["predict"]:
+            raise RuntimeError(
+                "The model for prediction has not been prepared."
+            )
+        self._mode = "predict"
+        self._engine.to_mode("predict")
+
+    def _make_feeds(self, data_list):
+        if (
+            self._mode not in self._feed_name_list
+            or self._feed_name_list[self._mode] == []
+        ):
+            feed_list = self._engine.get_feed_list()
+            self._feed_name_list[self._mode] = [var.name for var in feed_list]
+        feed_name_list = self._feed_name_list[self._mode]
+        if len(feed_name_list) != len(data_list):
+            raise ValueError(
+                "The input data and feed_list are not consistent."
+                "The model takes %s as input" % (str(feed_name_list))
+            )
+        return dict(zip(feed_name_list, data_list))
+
+    def __call__(self, *args):
+        if self._mode is None:
+            raise ValueError("Please call train()/eval()/predict() first.")
```

Comment:
default is train

Reply:
Done, default mode is set according to the init args.